modules/cudev/include/opencv2/cudev/functional/functional.hpp

```diff
@@ -668,6 +668,33 @@ template <typename T, typename D> struct saturate_cast_func : unary_function<T,
     }
 };
 
+// Convert Fp16 dummy
+template <typename T, typename D> struct saturate_cast_fp16_func : unary_function<T, D>
```

Comment:
Thank you. I removed the basic template function.  I fixed similar part below, too.